pkg/core/klinedriver/tick_klinedriver.go

```diff
@@ -0,0 +1,180 @@
+package klinedriver
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/c9s/bbgo/pkg/types"
+	log "github.com/sirupsen/logrus"
+)
+
+type TickKLineDriver struct {
+	symbol        string
+	tickInterval  types.Interval
+	intervals     map[types.Interval]struct{}
+	intervalSlice []types.Interval
+
+	tradeTimeLocation *time.Location
+	kLineEmitter      KLineEmitter
+
+	mu           sync.Mutex
+	builder      *KLineBuilder
+	running      bool
+	tradesBuffer []*types.Trade
+}
+
+type KLineEmitter interface {
+	EmitKLine(types.KLine)
+	EmitKLineClosed(types.KLine)
+}
+
+func NewTickKLineDriver(symbol string, tickInterval types.Interval) *TickKLineDriver {
```

Comment:
tickInterval should be time.Duration ?

Reply:
I feel `types.Interval` is more appropriate data type here since `time.Duration` can be negative, which is logically not suitable for tick interval.
On the other hand, `types.Interval` must have positive duration (and will panic if the user passed invalid value).

Reply:
well actually types.Interval only maps to the specific supported intervals, it could confuse users since tickInterval is only for time.Ticker 

tickerInterval here is usually set by human, which means when it's invalid, it's a logical exception 

and you don't need to worry about the negative number, time.Ticker always checks if this value is valid

Reply:
BTW: The main purpose of types.Interval is for parsing the config interval string

Reply:
I see.
I will update it as `time.Duration`.

Reply:
thanks